src/resolve/resolved-dnstls-gnutls.c

```diff
@@ -67,6 +67,12 @@ int dnstls_stream_connect_tls(DnsStream *stream, DnsServer *server) {
                 gnutls_session_set_verify_cert2(gs, &stream->dnstls_data.validation, 1, 0);
         }
 
+        if (server->server_name) {
+                r = gnutls_server_name_set(gs, GNUTLS_NAME_DNS, server->server_name, strlen(server->server_name));
+                if (r < 0)
+                        return log_debug_errno(SYNTHETIC_ERRNO(EINVAL), "Failed to set server name: %s", gnutls_strerror(r));
+        }
```

Comment:
hmm, error handling in this function is a bit borked, but i guess htat's not the fault of this PR... the function returns two kinds of errors: unix errno errors and gnutls errors in the same return value... urks this should be fixed, but i guess doesn't have to be fixed by your PR...